src/fsharp/TypeChecker.fs

```diff
@@ -7635,6 +7635,11 @@ and TcComputationExpression cenv env overallTy mWhole interpExpr builderTy tpenv
         let mBuilderVal = mBuilderVal.MakeSynthetic()
         SynExpr.Lambda (false,false,SynSimplePats.SimplePats ([mkSynSimplePatVar false (mkSynId mBuilderVal builderValName)],mBuilderVal), runExpr, mBuilderVal)
 
+    let env =
+        match comp with     
+        | SynExpr.YieldOrReturn _ -> { env with eContextInfo = ContextInfo.ReturnInComputationExpression }
```

Comment:
Will misuses of "yield" now report an error about "return"?  Test cases for that?